### ISSUE TITLE
RotatingProviderWithChangingKeysSpec: Futures timed out after [8 seconds]

### ISSUE THREAD

**Issue by raboof** [akka/akka#29331]
https://jenkins.akka.io:8498/job/akka-nightly-jdk8/7797/consoleFull

Likely the same problem as #29298 ?

```
[info] akka.remote.artery.tcp.ssl.RotatingProviderWithChangingKeysSpec *** ABORTED *** (13 seconds, 656 milliseconds)
[info]   java.util.concurrent.TimeoutException: Futures timed out after [8 seconds]
[info]   at scala.concurrent.impl.Promise$DefaultPromise.ready(Promise.scala:259)
[info]   at scala.concurrent.impl.Promise$DefaultPromise.result(Promise.scala:263)
[info]   at scala.concurrent.Await$.$anonfun$result$1(package.scala:220)
[info]   at scala.concurrent.BlockContext$DefaultBlockContext$.blockOn(BlockContext.scala:57)
[info]   at scala.concurrent.Await$.result(package.scala:146)
[info]   at akka.remote.artery.tcp.ssl.RotatingKeysSSLEngineProviderSpec.$anonfun$beforeTermination$1(RotatingKeysSSLEngineProviderSpec.scala:275)
[info]   at akka.remote.artery.tcp.ssl.RotatingKeysSSLEngineProviderSpec.$anonfun$beforeTermination$1$adapted(RotatingKeysSSLEngineProviderSpec.scala:272)
```

**Comment by chbatey** [comment]
https://jenkins.akka.io:8498/job/akka-multi-node/4122/testReport/junit/akka.remote.artery.tcp.ssl/RotatingProviderWithChangingKeysSpec/_It_is_not_a_test_it_is_a_sbt_testing_SuiteSelector_/

**Comment by raboof** [comment]
https://jenkins.akka.io:8498/job/akka-nightly-jdk8/7804/testReport/junit/akka.remote.artery.tcp.ssl/RotatingProviderWithChangingKeysSpec/_It_is_not_a_test_it_is_a_sbt_testing_SuiteSelector_/

**Comment by chbatey** [comment]
https://jenkins.akka.io:8498/job/akka-nightly-jdk8/7889/testReport/junit/akka.remote.artery.tcp.ssl/RotatingProviderWithChangingKeysSpec/_It_is_not_a_test_it_is_a_sbt_testing_SuiteSelector_/

**Comment by jrudolph** [comment]
https://jenkins.akka.io:8498/job/pr-validator-per-commit-jenkins/20541/testReport/junit/akka.remote.artery.tcp.ssl/RotatingProviderWithStaticKeysSpec/_It_is_not_a_test_it_is_a_sbt_testing_SuiteSelector_/

**Comment by seglo** [comment]
https://jenkins.akka.io:8498/job/akka-multi-node/4178/testReport/akka.remote.artery.tcp.ssl/RotatingProviderWithStaticKeysSpec/_It_is_not_a_test_it_is_a_sbt_testing_SuiteSelector_/

**Comment by seglo** [comment]
https://jenkins.akka.io:8498/job/akka-nightly-jdk8/7966/testReport/akka.remote.artery.tcp.ssl/RotatingProviderWithStaticKeysSpec/

**Comment by johanandren** [comment]
https://jenkins.akka.io:8498/job/akka-nightly-jdk8/7984/testReport/junit/akka.remote.artery.tcp.ssl/RotatingProviderWithChangingKeysSpec/_It_is_not_a_test_it_is_a_sbt_testing_SuiteSelector_/

**Comment by johanandren** [comment]
https://jenkins.akka.io:8498/job/akka-nightly-jdk8/7992/testReport/junit/akka.remote.artery.tcp.ssl/RotatingProviderWithStaticKeysSpec/_It_is_not_a_test_it_is_a_sbt_testing_SuiteSelector_/

**Comment by jrudolph** [comment]
https://jenkins.akka.io:8498/job/akka-nightly-jdk8/7995/testReport/akka.remote.artery.tcp.ssl/RotatingProviderWithChangingKeysSpec/_It_is_not_a_test_it_is_a_sbt_testing_SuiteSelector_/

**Comment by jrudolph** [comment]
https://jenkins.akka.io:8498/job/akka-nightly-jdk8/8008/testReport/junit/akka.remote.artery.tcp.ssl/RotatingProviderWithStaticKeysSpec/_It_is_not_a_test_it_is_a_sbt_testing_SuiteSelector_/

**Comment by patriknw** [comment]
https://jenkins.akka.io:8498/job/akka-multi-node/4204

**Comment by ignasi35** [comment]
The flakiness is a timeout when shutting down actor systems that joined other actor systems using akka remote.
The flaky tests (there's 2) run up to 5 actor systems connected between them in two separate groups: a group of 3 and a group of 2.
The flaky implementation terminates the actor systems sequentially causing the last-to-terminate actor system to experience failures when peers terminate. I wonder if stopping the actor systems in parallel would be more graceful.

**Comment by raboof** [comment]
> The flaky implementation terminates the actor systems sequentially causing the last-to-terminate actor system to experience failures when peers terminate

That kind of sounds like a bug, since the systems terminate gracefully, right? If we can remove the instability first and look into the termination problem in another PR that would be good, though.

> I wonder if stopping the actor systems in parallel would be more graceful/faster.

That sounds even more chaotic, but perhaps worth a try :smile:

**Comment by seglo** [comment]
https://jenkins.akka.io:8498/job/akka-nightly-jdk8/8027/consoleFull

**Comment by patriknw** [comment]
I agree that we should look into why it takes long time to terminate the systems. Artery has `flush-wait-on-shutdown = 2 s`, and if it can't complete the flush (ping-pong message) it will timeout and shutdown anyway. Maybe that is the reason? Strange that it doesn't happen every time then when they were shutdown sequentially? Something different when enabling TLS?

**Comment by raboof** [comment]
(not excluded and not seen for ~half a year, closing)
